File: java_patterns/src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/Elements/ConcreteElementB.java
package Basic_Patterns.Behavioural_Patterns.Visitor.Concept.code.Elements;

import Basic_Patterns.Behavioural_Patterns.Visitor.Concept.code.Visitors.Visitor;

public class ConcreteElementB implements Element {
    @Override
    public void accept(Visitor visitor) {
        visitor.visit(this);
    }

    public String specialMethodOfConcreteElementB() {
        return "B";
    }
}

File: typescript/src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/elements/ConcreteElementB.ts
import {Visitor} from "src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/visitors/Visitor";
import {Element} from "src/Basic_Patterns/Behavioural_Patterns/Visitor/Concept/code/elements/Element";

class ConcreteElementB implements Element {
    public accept(visitor: Visitor): void {
        visitor.visitConcreteElementB(this);
    }

    public specialMethodOfConcreteElementB(): string {
        return 'B';
    }
}

export { ConcreteElementB };